spot-ml/src/main/scala/org/apache/spot/dns/DNSSuspiciousConnectsAnalysis.scala

```diff
@@ -62,38 +62,45 @@ object DNSSuspiciousConnectsAnalysis {
     * @param logger
     */
   def run(config: SuspiciousConnectsConfig, sparkSession: SparkSession, logger: Logger,
-          inputDNSRecords: DataFrame): SuspiciousConnectsAnalysisResults = {
+          inputDNSRecords: DataFrame): Option[SuspiciousConnectsAnalysisResults] = {
 
 
     logger.info("Starting DNS suspicious connects analysis.")
 
-    val dnsRecords = filterRecords(inputDNSRecords)
-      .select(InSchema: _*)
-      .na.fill(DefaultQueryClass, Seq(QueryClass))
-      .na.fill(DefaultQueryType, Seq(QueryType))
-      .na.fill(DefaultQueryResponseCode, Seq(QueryResponseCode))
+    logger.info("Validating schema...")
+    val schemaValidationResults = validateSchema(inputDNSRecords)
 
-    logger.info("Fitting probabilistic model to data")
-    val model =
-      DNSSuspiciousConnectsModel.trainModel(sparkSession, logger, config, dnsRecords)
+    if (schemaValidationResults.length > InputSchema.ResponseDefaultSize) {
```

Comment:
Yeah, it can be, just want to have the list of columns not working. I can return a tuple or case class with Pass/Fail and a Seq of messages. What do you think?